pkg/cmd/roachprod-microbench/model/math.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2025 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+//
+
+package model
+
+import (
+	"math"
+	"math/rand"
+	"sort"
+)
+
+const resampleCount = 1000
+const confidence = 0.95
+
+// calculateConfidenceInterval calculates the confidence interval for the ratio
+// of two sets of values. The confidence interval is calculated using a
+// bootstrap method.
+func calculateConfidenceInterval(newValues, oldValues []float64) ConfidenceInterval {
+	rng := rand.New(rand.NewSource(hash(newValues) + hash(oldValues)))
+	ratios := make([]float64, resampleCount)
+	resNew := make([]float64, len(newValues))
+	resOld := make([]float64, len(oldValues))
+	for i := 0; i < resampleCount; i++ {
+		resample(rng, newValues, resNew)
+		resample(rng, oldValues, resOld)
+		medOld := median(resOld)
+		if medOld == 0 {
```

Comment:
Not super familiar with the bootstrap method... wouldn't this introduce bias if this happens _frequently_?